Add cross join to Python Relational and PySpark API

khalidmammadov opened the pull request:
Adds missing cross join to PySpark and Relational API

khalidmammadov committed:
draft

khalidmammadov committed:
Latest

khalidmammadov committed:
Format fixes

khalidmammadov committed:
Format fix

khalidmammadov committed:
Fix enum gen

Mytherin commented on src/include/duckdb/common/enums/join_type.hpp:
```diff
@@ -27,10 +27,11 @@ enum class JoinType : uint8_t {
+	                 // so that the build side can be the smaller table
+	RIGHT_ANTI = 10, // RIGHT ANTI join is created by the optimizer when the children of an anti join need to be
+	                 // switched so that the build side can be the smaller table
+	CROSS = 11
```
Cross products are not handled as joins in DuckDB. Can we remove this from the internal enum again? We need to create a `CrossProductRelation` to create a cross product.

khalidmammadov replied:
Sure, will remove and come back

Mytherin reviewed (commented):
Thanks for the PR! One comment:

khalidmammadov committed:
Revert and fix

khalidmammadov committed:
Add cross function

khalidmammadov committed:
Formatting changes

khalidmammadov committed:
Empty

khalidmammadov committed:
Add cross to init

khalidmammadov commented:
@Mytherin I saw you also changed target branch to `feature` but that got many differences with `main`. And I based my branch on main originally. Not sure if you want me to use that as a base. Reverted to main for now

khalidmammadov commented:
Hey @Mytherin, any comments?

Mytherin commented:
Thanks for the changes! LGTM
